### ISSUE TITLE
[Request] Repeat operator

### ISSUE THREAD

**Issue by TORISOUP** [Cysharp/R3#115]
I request the operator (Repeat) that was in UniRx to subscribe again when OnCompleted.(I think  the operator will be that combines both Repeat and Retry, because OnCompleted in R3 is both a success and a failure.)

**Comment by neuecc** [comment]
Repeat/Retry has become difficult to use effectively because it no longer necessarily ends due to OnErrorResume.
Repeat is triggered when the source itself sends OnCompleted, and repeating in such cases leads to an infinite loop because the source has completed. 
(Of course, it's possible to stop using an operator like Take, which is safe.)
Additionally, unlike UniRx, all sources have the potential to completed, making them unsuitable for repeat.
Therefore, it becomes an operator that can't be effectively used unless the pipeline before and after is appropriate, ensuring that the source has not completed and is properly interspersed with OnErrorResumeFailure.

Moreover, the existence of OnErrorResume, along with async/await, has eliminated the need for repeating the entire pipeline.
For example, if you want to execute an asynchronous method tied to a click and repeat/retry based on the success or failure of that asynchronous method,

```csharp
button.OnClickAsObservable()
    .SelectAwait(async _ =>
    {
        RETRY:
        try
        {
            return await RequestAsync();
        }
        catch
        {
            goto RETRY;
        }
    });
```

This allows for retrying within the Select, a method not possible before async/await.
Creating a custom operator like `Retry(Func<ValueTask>)` might be a good idea. 
However, adding it as a standard feature is undesirable because it can be misleading (giving the illusion of retrying the entire pipeline), so I would prefer to avoid including it.

**Comment by TORISOUP** [comment]
One of the difficulties with Rx in the first place was handling OnError/OnCompleted, so I agree with you about the policy of not taking care of that at all (simplifying it).

However, as for Repeat, I proposed it because of applications like the following case.

This is because when you want to rebuild Observable for some reason(For example, replace the source of the Observable.FromEvent.), it is complicated to notify Observer of OnCompleted and then ask for resubscription. If this is the case, I would like to keep the Observer's subscription downstream while rebuilding only the Observable upstream. In such cases, I used Repeat in UniRx. I then made a suggestion to achieve the same thing in R3.

However, this behavior can be obtained by using `Multicast` to separate the subscription subject from the publisher source. Therefore, although I suggested it, it may not be necessary to stick with Repeat to that extent.
